src/commonMain/kotlin/fr/acinq/lightning/channel/Closing.kt

```diff
@@ -406,60 +404,60 @@ data class Closing(
-            remoteCommitPublished != null -> ChannelClosingType.Remote
-            nextRemoteCommitPublished != null -> ChannelClosingType.Remote
-            futureRemoteCommitPublished != null -> ChannelClosingType.Remote
-            revokedCommitPublished.isNotEmpty() -> ChannelClosingType.Revoked
-            else -> ChannelClosingType.Other
-        }
-        additionalConfirmedTx?.let { confirmedTx ->
-            mutualClosePublished.firstOrNull { it.tx == confirmedTx }?.let {
-                txids += it.tx.txid
-                claimed += it.toLocalOutput?.amount ?: 0.sat
-            }
-        }
-        localCommitPublished?.let {
-            val confirmedTxids = it.irrevocablySpent.values.map { it.txid }.toSet()
-            val allTxs = listOfNotNull(it.claimMainDelayedOutputTx?.tx) +
-                    it.claimHtlcDelayedTxs.map { it.tx }
-            val confirmedTxs = allTxs.filter { confirmedTxids.contains(it.txid) }
-            if (confirmedTxs.isNotEmpty()) {
-                txids += confirmedTxs.map { it.txid }
-                claimed += confirmedTxs.map { it.txOut.first().amount }.sum()
-            }
-        }
-        listOfNotNull(
-            remoteCommitPublished,
-            nextRemoteCommitPublished,
-            futureRemoteCommitPublished
-        ).forEach {
-            val confirmedTxids = it.irrevocablySpent.values.map { it.txid }.toSet()
-            val allTxs = listOfNotNull(it.claimMainOutputTx?.tx) +
-                    it.claimHtlcTxs.mapNotNull { it.value?.tx }
-            val confirmedTxs = allTxs.filter { confirmedTxids.contains(it.txid) }
-            if (confirmedTxs.isNotEmpty()) {
-                txids += confirmedTxs.map { it.txid }
-                claimed += confirmedTxs.map { it.txOut.first().amount }.sum()
+    /**
+     * This method returns the closing transactions that should be persisted in a database. It should be
+     * called when the channel has been actually closed, so that we know those transactions are confirmed.
+     *
+     * @param additionalConfirmedTx additional confirmed transaction for mutual close; we need this for the
+     *      mutual close scenario because we don't store the closing tx in the channel state.
+     */
+    private fun storeChannelClosed(additionalConfirmedTx: Transaction): ChannelAction.Storage.StoreChannelClosed {
+
+        /** Helper method to extract a [OutgoingPayment.ClosingTxPart] from a list of transactions, given a set of confirmed tx ids. */
+        fun extractClosingTx(
+            confirmedTxids: Set<ByteVector32>,
+            txs: List<Transaction>,
+            closingType: ChannelClosingType
+        ): List<OutgoingPayment.ClosingTxPart> {
+            return txs.filter { confirmedTxids.contains(it.txid) }.map {
+                OutgoingPayment.ClosingTxPart(
+                    id = UUID.randomUUID(),
+                    txId = it.txid,
+                    claimed = it.txOut.first().amount,
```

Comment:
@sstone Is it correct to only get the first output here? This is what was done previously so that part has not changed, but I'd like to be sure we are storing the correct amount.